packages/client/hmi-client/src/services/data.ts

```diff
@@ -251,72 +130,98 @@ const filterDatasets = (
 	return term.length > 0 ? uniqBy(finalDatasets, ID) : allDatasets;
 };
 
-const getDatasets = async (term: string, datasetSearchParam?: DatasetSearchParams) => {
+const getAssets = async (
+	term: string,
+	resourceType: ResourceType,
+	searchParam?: ModelSearchParams | DatasetSearchParams
+) => {
 	const results = {} as FullSearchResults;
 
-	//
-	// fetch list of datasets from the HMI server
-	//
-	const datasetsList = (await DatasetService.getAll()) || ([] as Dataset[]);
-
-	// TEMP: add "type" field because it is needed to mark these resources as datasets
-	// FIXME: dependecy on type dataset should be removed and another "sub-system" or "result-type"
-	//        should be added for other resource types
-	const allDatasets = datasetsList.map((d) => ({
-		...d,
-		temporalResolution: d.temporal_resolution,
-		geospatialResolution: d.geospatial_resolution,
-		simulationRun: d.simulation_run,
-		type: 'dataset'
-	}));
+	// fetch list of model or datasets data from the HMI server
+	let assetList: Model[] | Dataset[] = [];
+	let projectAssetType: ProjectAssetTypes;
+
+	switch (resourceType) {
+		case ResourceType.MODEL:
+			assetList = (await getAllModelDescriptions()) || ([] as Model[]);
+			projectAssetType = ProjectAssetTypes.MODELS;
+			break;
+		case ResourceType.DATASET:
+			assetList = (await DatasetService.getAll()) || ([] as Dataset[]);
+			projectAssetType = ProjectAssetTypes.DATASETS;
+			break;
+		default:
+			return results; // error or make new resource type compatible
+	}
 
-	let conceptFacets = await getConceptFacets([ProjectAssetTypes.DATASETS]);
+	// TEMP: add "type" field because it is needed to mark these resources as models or datasets
+	// FIXME: dependency on type model should be removed and another "sub-system" or "result-type"
+	//        should be added for datasets and other resource types
+	const allAssets = assetList.map((a) => ({
+		...a,
+		temporalResolution: a?.temporal_resolution, // Dataset attribute
+		geospatialResolution: a?.geospatial_resolution, // Dataset attribute
+		simulationRun: a?.simulation_run, // Dataset attribute
+		type: resourceType
+	}));
 
-	const datasetResults = filterDatasets(allDatasets, conceptFacets, term);
+	// first get un-filtered concept facets
+	let conceptFacets = await getConceptFacets([projectAssetType]);
 
-	// FIXME: this client-side computation of facets from "dataset" data should be done
+	// FIXME: this client-side computation of facets from "models" data should be done
 	//        at the HMI server
 	//
 	// This is going to calculate facets aggregations from the list of results
-	const datasetFacets = await getDatasetFacets(datasetResults, conceptFacets);
+	let assetResults: Model[] | Dataset[];
+	let assetFacets: Facets;
+
+	switch (resourceType) {
```

Comment:
There's a few places where we do a switch and then call a method that is basically identical whether it's for models or datasets. I am tempted to say that where this happens it can be combined into one method (e.g. `getAssetFacets` instead of `getModelFacets` and `getDatasetFacets`, making use of function overloading. 

However, if all this logic is going to be moved to server side then it's probably not worth it? What do you think @YohannParis

Reply:
Yes there will still be some repetition even after I do more clean up. The `getFacets` functions will be moved to the server side later.